rust-symcrypt/src/ecdh.rs

```diff
@@ -0,0 +1,270 @@
+/* EcDh functions. For further documentation please refer to symcrypt.h */
+
+use crate::curve_type::*;
+use crate::errors::SymCryptError;
+use std::vec;
+use symcrypt_sys;
+
+/// EcDhKey is a wrapper around symcrypt_sys::PSYMCRYPT_ECKEY, this is to let rust handle the free'ing of this pointer
+/// EcDhKey must be allocated after EcDhExpandedCurve and free'd before EcDhExpandedCurve is free'd
+struct EcDhKey {
+    inner: symcrypt_sys::PSYMCRYPT_ECKEY,
+    expanded_curve: EcDhExpandedCurve,
+}
+
+struct EcDhExpandedCurve(symcrypt_sys::PSYMCRYPT_ECURVE);
```

Comment:
>EcDhExpandedCurve [](http://example.com/codeflow?start=7&length=17)

Why "expanded"? Also, `PSYMCRYPT_ECURVE` is used for other elliptic curve algorithms, so I'm not sure it makes sense to wrap it with a structure that is named specifically for ECDH.